react-catalog/my-app/src/Catalog.tsx

```diff
@@ -0,0 +1,54 @@
+import { useEffect, useState } from 'react';
+import { readCatalog } from './read';
+import { type Product } from './data';
+import { Link } from 'react-router-dom';
+
+export function Catalog() {
+  const [products, setProducts] = useState<Product[]>();
+  useEffect(() => {
+    async function productCall() {
+      try {
+        const catalog = await readCatalog();
+        setProducts(catalog);
+      } catch (error) {
+        console.error(error);
+      }
+    }
+    productCall();
+  }, [products]);
+  return (
+    <>
+      <h1 className="border-b-2 w-full text-[40px] font-semibold">Catalog</h1>
+      <div className="flex flex-wrap">
+        {products?.map((product) => (
+          <ItemCard key={product.productId} product={product} />
+        ))}
+      </div>
+    </>
+  );
+}
+
+type ItemCardProp = {
+  product: Product;
+};
+
+function ItemCard({ product }: ItemCardProp) {
+  return (
+    <Link
+      to={`details/${product.productId}`}
+      className="mx-8 my-4 w-[240px] h-[380px] border-2 rounded-lg">
+      <div className="w-[236px] h-[236px] flex justify-center items-center">
+        <img
+          src={product.imageUrl}
+          className="max-w-[236px] max-h-[236px]"></img>
+      </div>
+      <div className="m-4">
+        <h3 className="text-[14px] font-semibold">{product.name}</h3>
+        <h6 className="text-gray-400 text-[12px] font-semibold">{`$${String(
```

Comment:
The `toDollars` function was provided in the `lib` folder so that you can use it here. It's actually much easier to display dollars and cents than what you've done here -- just use `.toFixed(2)`.